### PR TITLE
WIP: Add PETSc test fixture

### PR DESCRIPTION
**Pull request by fsimonis**
This PR adds the fixture `PETScTestFixture`, which restricts the communicator to a given amount of ranks and then both initializes and finalizes PETSc.

This has two effects:
1. Every PetRBF test now runs in a fresh environment, which makes debugging saner as the test order does not matter any more.
2. The test `main` does not have to initialize or finalize PETSc any more.

This is a first step of #597

### CODE REVIEW

**Comment by fsimonis** [comment]
WIP as I have a more consistent method in mind

**Comment by BenjaminRueth** [comment]
Is this related to https://github.com/precice/systemtests/issues/139?

File: src/testing/Fixtures.hpp [hunk 1]
```diff
@@ -105,4 +107,35 @@ struct SplitParticipantsFixture {
 
 };
 
+#ifndef PRECICE_NO_PETSC
+
+/** Fixture to initialize and finalize PETSc for a given amount of ranks.
```
Comment (uekerman) [inline src/testing/Fixtures.hpp]:
```suggestion
/** Fixture to initialize and finalize PETSc in a unit test for a given amount of ranks.
```

**Comment by uekerman** [comment]
> Is this related to [precice/systemtests#139](https://github.com/precice/systemtests/issues/139)?

No, it's a pure refactoring to improve maintainability of tests, see #597.

**Comment by fsimonis** [comment]
As it turns out, this PR is a fix for the problem, not a solution.

**Pull request closed without merging**